### ISSUE TITLE
4.7 not working with Minecraft 1.1.7

### ISSUE THREAD

**Issue by Thenaikan** [toolbox4minecraft/amidst#999]
Error attached.

![image](https://user-images.githubusercontent.com/67339081/122312496-53469f00-cee2-11eb-8cfe-94d8e37c17d4.png)

**Comment by JDHnz** [comment]
You need to upgrade to Java 16

**Comment by dgl-thrawn** [comment]
I have Java 16 JDK installed alongside Java 8 JRE. I'm able to run Minecraft 1.17 from both MultiMC and the regular launcher, but I get the same error message.

**Comment by Mikolsito** [comment]
I'm having the same issue too! :/

**Comment by JDHnz** [comment]
It is working for me. I downloaded "amidst-v4-7.jar", and am launching it with `java -jar amidst-v4-7.jar`. I only have the Microsoft build of OpenJDK 16 on my machine, the MSI for which can be found at https://docs.microsoft.com/en-nz/java/openjdk/download, which is currently https://aka.ms/download-jdk/microsoft-jdk-16.0.1.9.1-windows-x64.msi but this may be replaced with newer builds in the future.

I suspect the problem that everyone here is complaining about is with the "amidst-v4-7.exe" package, which appears to be specifically looking for JRE 8, as it gives this error when run on my machine:
![image](https://user-images.githubusercontent.com/18640585/122367665-bbfc4d00-cfb0-11eb-94e6-84f45cff32dc.png)

**Comment by dgl-thrawn** [comment]
> It is working for me. I downloaded "amidst-v4-7.jar", and am launching it with `java -jar amidst-v4-7.jar`. I only have the Microsoft build of OpenJDK 16 on my machine, the MSI for which can be found at https://docs.microsoft.com/en-nz/java/openjdk/download, which is currently https://aka.ms/download-jdk/microsoft-jdk-16.0.1.9.1-windows-x64.msi but this may be replaced with newer builds in the future.
> 
> I suspect the problem that everyone here is complaining about is with the "amidst-v4-7.exe" package, which appears to be specifically looking for JRE 8, as it gives this error when run on my machine:
> ![image](https://user-images.githubusercontent.com/18640585/122367665-bbfc4d00-cfb0-11eb-94e6-84f45cff32dc.png)

Thanks. Tried the jar version and it works fine.

**Comment by JDHnz** [comment]
I'm not a java dev, but I think the fix may simply be to update the jdk version referenced in src/main/resources/amidst/metadata.properties from 1.8 to 1.16

**Comment by Thenaikan** [comment]
![java](https://user-images.githubusercontent.com/67339081/122396796-5bd0c100-cf46-11eb-9d52-f4fa4c93019d.JPG)

Doesn;t work with the above Java version

**Comment by JDHnz** [comment]
@Thenaikan see my comment above for a way of getting it to work. You are correct that it is not possible to automatically update the version of Java using the built-in updater.

**Comment by anew742** [comment]
> 
> 
> It is working for me. I downloaded "amidst-v4-7.jar", and am launching it with `java -jar amidst-v4-7.jar`. I only have the Microsoft build of OpenJDK 16 on my machine, the MSI for which can be found at https://docs.microsoft.com/en-nz/java/openjdk/download, which is currently https://aka.ms/download-jdk/microsoft-jdk-16.0.1.9.1-windows-x64.msi but this may be replaced with newer builds in the future.
> 
> I suspect the problem that everyone here is complaining about is with the "amidst-v4-7.exe" package, which appears to be specifically looking for JRE 8, as it gives this error when run on my machine:
> ![image](https://user-images.githubusercontent.com/18640585/122367665-bbfc4d00-cfb0-11eb-94e6-84f45cff32dc.png)


Where exactly does one put those launch arguments?
I'm having the same issue as @Thenaikan and can't get it to work, even when using the .jar

**Comment by JDHnz** [comment]
From the command line. Probably the easiest way is to create a batch file in the same directory as the jar file and put the command in there.

**Comment by Tencryn** [comment]
Can confirm that this happens with the .exe despite having Java 16. The `.jar` file works fine though.

**Comment by andrewmichaellawson** [comment]
MSI download and jar version worked for me

**Comment by Mr-Technician** [comment]
> Can confirm that this happens with the `.exe` despite having Java 16. The `.jar` file works fine though.

Same here. Latest version of both the jdk and jre are installed.

**Comment by mlangas** [comment]
So here is a workaround.  Download the jar, not the exe.  Right click the icon and open the properties.  Make sure the "Open with" property is set to the MS version.  Mine was still trying to use version 1.8.

**Comment by Jupisoft111** [comment]
I had the same problem with the .EXE, but also with the .JAR. Luckily after installing the latest Java runtime 8 and JDK 16 now at least the .JAR is working now. And by the way I have several older versions installed, which still are required by some very old apps I still use nowadays and it still works. Hopefully soon the executable will also be fixed. Thanks a lot for all the help :-)

**Comment by JSmurf11** [comment]
Is the jdk version from oracle different from Microsoft?
Java(TM) SE Runtime Environment (build 1.8.0_291-b10) is what I have now and given the Microsoft jdk 1.16 from links above I'm planning to install the Microsoft version to run amidst

**Comment by JDHnz** [comment]
Nah, it was just easier to find than the correct one from Oracle

**Comment by JSmurf11** [comment]
> Nah, it was just easier to find than the correct one from Oracle

will installing microsoft-jdk-16.0.1.9.1-windows-x64.msi remove the current jdk that I'm using?

**Comment by JDHnz** [comment]
No

**Comment by MattLeBlizz** [comment]
> So here is a workaround. Download the jar, not the exe. Right click the icon and open the properties. Make sure the "Open with" property is set to the MS version. Mine was still trying to use version 1.8.

this worked for me without getting a different jdk, just asked to open with Java (TM) SE binary.

**Comment by AdoulM** [comment]
Any way to fix the problem in simpler words? EXE and JAR both don't work for me and I don't really understand what is being said here.

**Comment by Thenaikan** [comment]
> 
> 
> Any way to fix the problem in simpler words? EXE and JAR both don't work for me and I don't really understand what is being said here.
> 
> Edit: Nvm just downloaded microsoft-jdk-16.0.1.9.1-windows-x64.msi like someone above said and it worked.

I'm in the same boat and don't really have time to try to figure this all out. Just wait for the author to fix the exe - might be a while so try to be patient.

**Comment by lovchaudhary123** [comment]
> Error attached.
> 
> ![image](https://user-images.githubusercontent.com/67339081/122312496-53469f00-cee2-11eb-8cfe-94d8e37c17d4.png)

yes same

**Comment by fbalaguera** [comment]
Hello, try install this: [https://www.oracle.com/java/technologies/javase-jdk16-downloads.html](url) (Reboot when installed)
Then download and run the Amidst JAR file [https://github.com/toolbox4minecraft/amidst/releases](url) ([https://github.com/toolbox4minecraft/amidst/releases/download/v4.7/amidst-v4-7.jar](url))
That should be enough

**Comment by miketoles** [comment]
Installing JDK16 doesn't work on Mac. <img width="1670" alt="Screen Shot 2021-07-18 at 10 00 48 AM" src="https://user-images.githubusercontent.com/23155671/126074094-c577b6ed-04be-4a66-85aa-54f7a39e01f8.png">
